webserver/src/handlers/stripe_handler.rs

```diff
@@ -0,0 +1,56 @@
+use actix_web::{HttpResponse, web, post, Responder};
+use crate::services::stripe_service;
+use crate::models::payment::{CreatePaymentRequest, CheckPaymentStatusResponse, PaymentIntentResponse, ConfirmPaymentRequest};
+use reqwest::Client;
+use serde_json::json;
+
+
+pub async fn create_payment(req: web::Json<CreatePaymentRequest>) -> HttpResponse {
+    let CreatePaymentRequest { amount, ref currency, ref payment_method_types }  = req.0; 
+    // Extract data from the request and pass it to the service layer
+    match stripe_service::create_payment(req.amount, &req.currency).await {
+        Ok(payment_intent) => HttpResponse::Ok().json(payment_intent),
+        Err(e) => {
+            eprintln!("Error creating payment: {}", e);
+            HttpResponse::InternalServerError().json("Failed to create payment")
+        }

```

Comment:
Please use the correct error message. Should be [502 Bad Gateway](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/502)